include/transport/time_queue.h

```diff
@@ -377,17 +412,54 @@ namespace qtransport {
         size_t size() const { return _queue.size() - _queue_index; }
         bool empty() const { return (_queue.empty() || _queue_index >= _queue.size()); }
 
-      private:
+    private:
+        /**
+         * @brief Checks if currently in allowed TTL spike
+         *
+         * @note This should be called only when the front/pop would expire an
+         *      object due to TTL being expired.
+         *
+         * @details Latency spike
+         *
+         * @retun True if currently in spike duration and the TTL should be allowed/not expired
+         *      False if not in spike duration and the object should be expired.
+         */
+         bool in_spike(tick_type current_tick) {
+             if (_spike_period_interval == 0) { // zero means disabled, so return fast as possible
+                 // Disabled
+                 return false;
+             }
+
+             if (_spike_duration_end && current_tick <= _spike_duration_end) {
+                 // Within the allowed spike duration
+                 return true;
+             }
+
+             if (current_tick > _spike_period_end) {
+                 // Start new spike period and allow since this is a new starting spike duration
+                 _spike_period_end = current_tick + _spike_period_interval;
+                 _spike_duration_end = current_tick + _spike_duration;
+
+                 return true;
+             }
+
+             return false;
+         }
         /**
          * @brief Based on current time, adjust and move the bucket index with time
          *        (sliding window)
          *
          * @returns Current tick value at time of advance
          */
-        inline tick_type advance()
+        tick_type advance()
         {
             _timer->get_ticks(Duration_t(_interval), _timer_ctx);
 
+            if (_queue_index && _queue.size() >= _total_buckets && _queue_index >= _queue.size()) {
```

Comment:
```cpp
_queue.size() >= _total_buckets
```
This could and absolutely should be true and okay though? The number of buckets does not correlate with the size of of the queue. For instance, every item in the queue could be in bucket 0, let's say we have 5 buckets and a queue size of 100. That's completely fine, so why check this?

As well, let's we have 2 items in the queue, and 5 buckets,  but the queue_index is 2. We've run out the queue, but the number of elements isn't greater than the buckets, so we don't clear (even though we 100% should, since we've consumed the items already).

Reply:
I think this code should probably go back where it was as it was, this adds some potentially unnecessary checks (queue may be cleared by a jump on line 429). An I'm still not so sure this check against total buckets is correct